### PR TITLE
Added  Unidex to perpetual_trades

### PR DESCRIPTION
**Pull request by ARDev097**
Working on perps trade file for Unidex and Kwenta.

### CODE REVIEW

File: models/unidex/optimism/unidex_optimism_perpetual_trades.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{{ config(
+    
+    alias = 'unidex_perpetual_trades',
```
Comment (Hosuke) [inline models/unidex/optimism/unidex_optimism_perpetual_trades.sql:3]:
```suggestion
    alias = 'perpetual_trades',
```

File: models/unidex/optimism/unidex_optimism_perpetual_trades.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    schema= 'unidex_optimism_perpetual',
```
Comment (Hosuke) [inline models/unidex/optimism/unidex_optimism_perpetual_trades.sql:4]:
```suggestion
    schema= 'unidex_optimism',
```

**Comment by Hosuke** [comment]
This PR may be merged after #5040, as they shared the common schema and source files.

**Comment by ARDev097** [comment]
Hello @Hosuke , the PR #5040 has been merged. Please check it and let me know whether anything is required from my end for this PR or not.

**Comment by Hosuke** [comment]
> Hello @Hosuke , the PR #5040 has been merged. Please check it and let me know whether anything is required from my end for this PR or not.

Hi @ARDev097, you can resolve conflicts first and sync this PR with the main branch.

File: models/unidex/unidex_perpetual_trades.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{{ config(
+    
+    alias = 'perpetual_trades',
+    schema = 'unidex_perpetual',
```
Comment (Hosuke) [inline models/unidex/unidex_perpetual_trades.sql:4]:
```suggestion
    schema = 'unidex',
    alias = 'perpetual_trades',
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ARDev097 

[Sample unidex_v1_optimism_perpetual_trades](https://dune.com/queries/3330304)
[Sample unidex_v2_optimism_perpetual_trades](https://dune.com/queries/3330306)
[Sample unidex_v3_optimism_perpetual_trades](https://dune.com/queries/3330308)

File: models/unidex/optimism/unidex_optimism_perpetual_trades.sql [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{{ config(
+    
+    alias = 'perpetual_trades',
+    schema= 'unidex_optimism',
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                    "project",
+                                    "unidex",
+                                    \'["ARDev097"]\') }}'
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_optimism_perpetual_trades.sql:8]:
```suggestion

```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	alias = 'perpetual_trades',
+	partition_by = ['block_month'],
+	materialized = 'incremental',
+	file_format = 'delta',
+	incremental_strategy = 'merge',
+	unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                "project",
+                                "unidex",
+                                \'["ARDev097"]\') }}'
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:12]:
```suggestion

```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+		,0 AS fee
+		,contract_address
+		,evt_tx_hash
+		,evt_index
+		,evt_block_time
+		,evt_block_number
+		,'1' AS version
+	FROM {{ source('unidex_optimism', 'trading_evt_NewOrder') }}
+	{% if is_incremental() %}
+	WHERE evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:34]:
```suggestion
	WHERE {{ incremental_predicate('evt_block_time') }}
```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+		,0 AS fee
+		,contract_address
+		,evt_tx_hash
+		,evt_index
+		,evt_block_time
+		,evt_block_number
+		,'1' AS version
+	FROM {{ source('unidex_optimism', 'trading_evt_ClosePosition') }}
+	{% if is_incremental() %}
+	WHERE evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:54]:
```suggestion
	WHERE {{ incremental_predicate('evt_block_time') }}
```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+	,tx."to" AS tx_to
+	,perps.evt_index
+FROM perps
+INNER JOIN {{ source('optimism', 'transactions') }} AS tx
+	ON perps.tx_hash = tx.hash
+	AND perps.block_number = tx.block_number
+	{% if not is_incremental() %}
+	AND tx.block_time >= DATE '{{project_start_date}}'
+	{% endif %}
+	{% if is_incremental() %}
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:113]:
```suggestion
	{% else %}
```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+	AND tx.block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
```
Comment (jeff-dude) [inline models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:114]:
```suggestion
	AND {{ incremental_predicate('tx.block_time') }}
```

File: models/unidex/unidex_perpetual_trades.sql [hunk 10]
```diff
@@ -0,0 +1,46 @@
+{{ config(
+    schema = 'unidex',
+    alias = 'perpetual_trades',
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                    "project",
+                                    "unidex",
+                                    \'["ARDev097"]\') }}'
```
Comment (jeff-dude) [inline models/unidex/unidex_perpetual_trades.sql:7]:
```suggestion

```

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql (hunk 5 above)
Comment (jeff-dude) [reply models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:12]: plz apply to all version spells

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql (hunk 6 above)
Comment (jeff-dude) [reply models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:34]: plz apply to all version spells

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql (hunk 7 above)
Comment (jeff-dude) [reply models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:54]: plz apply to all version spells

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql (hunk 8 above)
Comment (jeff-dude) [reply models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:113]: plz apply to all version spells

File: models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql (hunk 9 above)
Comment (jeff-dude) [reply models/unidex/optimism/unidex_v1_optimism_perpetual_trades.sql:114]: plz apply to all version spells

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**